AWSAppSyncClient/Internal/AWSAppSyncClientInfo.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+import Foundation
+
+public protocol AWSAppSyncClientInfoProviding {
+    var apiUrl: String { get }
+    var region: String { get }
+    var authType: String { get }
+    var apiKey: String { get }
+}
+
+/**
+ * Configuration for AWSAppSyncClient as read from `awsconfiguration.json`
+ */
+public class AWSAppSyncClientInfo: AWSAppSyncClientInfoProviding {
+    private struct ConfigFileKeys {
+        static let root = "AppSync"
+        static let apiURL = "ApiUrl"
+        static let region = "Region"
+        static let authMode = "AuthMode"
+        static let apiKey = "ApiKey"
+    }
+
+    public let apiUrl: String
+    public let region: String
+    public let authType: String
```

Comment:
I suggest here use `AuthType` instead of just `String`. And also rename `AuthType` into `AWSAppSyncAuthType`.

Reply:
I agree with this change: Since `AuthType` was an internal type, renaming without a breaking change is fine. Since I'm promoting it to public visibility by making it the type of one of the configurations, namespacing it is also appropriate.

Reply:
Er, I meant to say, "I agree with this suggestion", or more succinctly, "Thanks, will do!" :)